src/trie.js

```diff
@@ -2,6 +2,16 @@ var marexandre;
 (function (marexandre) {
   'use strict';
 
+  /**
+   * hasChildrenWithValue returns if node of trie has a child
```

Comment:
Oh, you're right, it look for a single child. Then could you rename the function to `hasChildWithValue` ?
Also, the comment is a bit vague. Maybe more `hasChildrenWithValue returns true if the node of a trie has a child with the requested value`.